charts/tfy-otel-collector/templates/_helpers.tpl

```diff
@@ -0,0 +1,173 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "tfy-otel-collector.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+  Create a default fully qualified app name.
+  We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+  If release name contains chart name it will be used as a full name.
+  */}}
+{{- define "tfy-otel-collector.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+  Create chart name and version as used by the chart label.
+  */}}
+{{- define "tfy-otel-collector.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+  Common labels
+  */}}
+{{- define "tfy-otel-collector.labels" -}}
+helm.sh/chart: {{ include "tfy-otel-collector.chart" . }}
+{{- range $name, $value := .Values.commonLabels }}
+{{ $name }}: {{ tpl $value $ | quote }}
+{{- end }}
+{{ include "tfy-otel-collector.selectorLabels" . }}
+{{- if .Values.image.tag }}
+app.kubernetes.io/version: {{ .Values.image.tag | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+{{/*
+Annotations
+*/}}
+{{- define "tfy-otel-collector.annotations" -}}
+{{- if .Values.commonAnnotations }}
+  {{- toYaml .Values.commonAnnotations }}
+{{- else }}
+{}
+{{- end }}
+{{- end }}
+
+{{/*
+  Selector labels
+  */}}
+{{- define "tfy-otel-collector.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "tfy-otel-collector.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+  Create the name of the service account to use
+  */}}
+{{- define "tfy-otel-collector.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "tfy-otel-collector.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+ServiceAccount Annotation
+*/}}
+{{- define "tfy-otel-collector.serviceAccount.annotations" -}}
+{{- if .Values.serviceAccount.create }}
+  {{- toYaml .Values.serviceAccount.annotations }}
+{{- else if .Values.commonAnnotations }}
+  {{- toYaml .Values.commonAnnotations }}
+{{- else }}
+  {}
+{{- end }}
+{{- end }}
+
+{{/*
+  Parse env from template
+  */}}
+{{- define "tfy-otel-collector.parseEnv" -}}
+{{ tpl (.Values.env | toYaml) . }}
+{{- end }}
+
+{{/*
+  Create the env file
+  */}}
+{{- define "tfy-otel-collector.env" }}
+{{- range $key, $val := (include "tfy-otel-collector.parseEnv" .) | fromYaml }}
+{{- if and $val (contains "${k8s-secret" ($val | toString)) }}
```

Comment:
If we are using this way then we can freely use `${k8s-secret/secretName/KeyName}` in the clickhousePassword

Reply:
Passing `${k8s-secret/secretName/KeyName}` in the values file doesn't seem to work. For now only passing in the actual password during deployment works.